STYLE.md

```diff
@@ -0,0 +1,336 @@
+# Dafny Standard Library Style Guide
+
+This style guide provides coding conventions for the Dafny Standard Library code.
+
+*This documentation is a modification of the Dafny Style Guide and is still in progress.*
+
+## Naming Convention
+
+Any **classes**, **modules**, **datatypes**, and **newtypes** are named with `PascalCase`.
+
+    Module MyModule {
+        ...
+    }
+    
+Any **lemmas**, **predicates**, **functions**, **methods** and **variables** are named with `snake_case`.
+
+    method find_index(arr: seq<int>, k: int)
+        ...
+        
+        
+The **lemma** keyword indicates a ghost method used for proof purposes. Any **lemma** names should be prefixed with `lemma_`.
```

Comment:
We've found it to be a useful convention, since Dafny's various UIs don't make it easy to distinguish ghost and non-ghost code.  In particular, when reading through a complex implementation, it's helpful to be able to see at a glance which calls can potentially mutate state, and which are only being called for proof purposes.

Reply:
I figured that was the motivation. I would love to make ghost state/code easily distinguishable in the UIs. I've noticed that when folks used to more mainstream engineering first start learning Dafny, they often freak out about what looks like extremely expensive implementations, not realizing that the logic in question is erased by the compilers.